tests/test_post_to_studio.py

```diff
@@ -464,3 +464,38 @@ def test_post_to_studio_skip_if_no_repo_url(
         live.next_step()
 
     assert mocked_post.call_count == 0
+
+
+def test_post_to_studio_repeat_step(tmp_dir, mocked_dvc_repo, mocked_studio_post):
+    live = Live()
+
+    foo_path = (Path(live.plots_dir) / Metric.subfolder / "foo.tsv").as_posix()
+
+    mocked_post, _ = mocked_studio_post
+
+    live.step = 0
+    live.log_metric("foo", 1)
+    live.sync()
```

Comment:
is `live.sync()` considered public / documented? or is it an internal implementation?

Reply:
We haven't had a good reason to document it, but it was written with the intent it could be made public if a use case for it arises. See the note in https://github.com/iterative/dvclive/pull/749#issue-2030974900.

Reply:
The test could also be written by using `next_step` followed by manually overriding the step value.

Reply:
okay, I don't have a strong preference here tbh. If we feel `sync` won't be refactored we can keep it this way (probably it's better vs patching the step).

do we need to add tests for when there are a few values logged at a different pace?

Reply:
anyways, folks, your call on all this ... I can review one more time from the technical point of view when you decide how to go